File: src-tauri/src/commands/refresh.rs
use tauri::State;

use crate::{
    state::HeldState,
    types::{Directory, FilePreview, Metadata, SortBy, SortOptions, SortOrder},
};

#[tauri::command]
pub fn refresh(state: State<HeldState>) -> Result<(Metadata, Vec<FilePreview>, Directory), String> {
    let state = match state.lock() {
        Ok(state) => state,
        Err(_) => return Err("Could not lock state".to_string()),
    };

    let metadata = Metadata::new(state.get_notes_dir().clone());
    let files = state.get_all_files(Some(SortOptions {
        by: SortBy::Name,
        order: SortOrder::Ascending,
    }));
    let directory = state.get_directory().map_err(|e| e.to_string())?;

    Ok((metadata, files, directory))
}

File: src/types/commands/refresh.ts
import { invoke } from "@tauri-apps/api";
import { FilePreview } from "../filePreview/filePreview";
import { Metadata } from "../metadata";
import { Directory } from "../files/directory";

export default async function refresh() {
  return invoke<[Metadata, FilePreview[], Directory]>("refresh", {});
}
